bluemix/helpers.js

```diff
@@ -10,218 +10,226 @@ var g = require('strong-globalize')();
+          when: !this.bluemix,
+        };
+
+        prompts.push(toolchainPrompt, dockerPrompt);
+
+        this.prompt(prompts).then(function(answers) {
+          this.appMemory = answers.appMemory;
+          this.appInstances = answers.appInstances;
+          this.appDomain = answers.appDomain;
+          this.appHost = answers.appHost;
+          this.appDiskQuota = answers.appDiskQuota;
+          this.enableDocker = answers.enableDocker || this.enableDocker;
+          this.enableToolchain = answers.enableToolchain ||
+            this.enableToolchain;
+          this.bluemix = this.enableManifest || this.enableDocker ||
+          this.enableToolchain;
+          done();
+        }.bind(this));
+      }.bind(this));
     };
 
-    var dockerPrompt = {
-      name: 'enableDocker',
-      message: g.f('Do you want to create Dockerfile?'),
-      type: 'confirm',
-      default: true,
-      when: !this.bluemix,
+    generateFiles() {
+      if (this.done) return;
+      if (this.bluemix) {
+        var bluemixOptions = {
+          destDir: this.destinationRoot(),
+          enableBluemix: this.enableBluemix,
+          enableManifest: this.enableManifest,
+          enableDocker: this.enableDocker,
+          enableToolchain: this.enableToolchain,
+        };
+        lbBm.generateBluemixFiles(bluemixOptions,
+          this.copy.bind(this), this.directory.bind(this));
+      }
     };
 
-    prompts.push(toolchainPrompt, dockerPrompt);
-
-    this.prompt(prompts).then(function(answers) {
-      this.appMemory = answers.appMemory;
-      this.appInstances = answers.appInstances;
-      this.appDomain = answers.appDomain;
-      this.appHost = answers.appHost;
-      this.appDiskQuota = answers.appDiskQuota;
-      this.enableDocker = answers.enableDocker || this.enableDocker;
-      this.enableToolchain = answers.enableToolchain || this.enableToolchain;
-      this.bluemix = this.enableManifest || this.enableDocker ||
-        this.enableToolchain;
-      done();
-    }.bind(this));
-  }.bind(this));
-};
-
-bluemix.generateFiles = function() {
-  if (this.done) return;
-  if (this.bluemix) {
-    var bluemixOptions = {
-      destDir: this.destinationRoot(),
-      enableBluemix: this.enableBluemix,
-      enableManifest: this.enableManifest,
-      enableDocker: this.enableDocker,
-      enableToolchain: this.enableToolchain,
+    promptDefaultServices() {
+      if (this.done) return;
+      if (this.enableDefaultServices) {
+        var prompts = [
+          {
+            name: 'enableAutoScaling',
+            message: g.f('Do you want to enable autoscaling?'),
+            type: 'confirm',
+          },
+          {
+            name: 'enableAppMetrics',
+            message: g.f('Do you want to enable appmetrics?'),
+            type: 'confirm',
+          },
+        ];
+
+        var self = this;
+        return this.prompt(prompts).then(function(answers) {
+          self.enableAutoScaling = answers.enableAutoScaling;
```

Comment:
AFAIK this can be `this.enableAutoScaling` instead as the function is bound to `this`.